system-addon/lib/PlacesFeed.jsm

```diff
@@ -239,6 +239,18 @@ class PlacesFeed {
     win.openLinkIn(action.data.url, "window", params);
   }
 
+  /**
+   * Currently used when bookmarking a TopStory to ensure it gets picked up in
+   * the Highlights section.
+   * If used in other cases, should ensure URLs actually have 0 visits.
```

Comment:
Oh. I was thinking this would just be called as part of BOOKMARK_URL, but this is good. This method description should describe what it does not how it's used. Also, the "should ensure URLs" is misleading as it seems to imply TopStory is making sure there's 0 visits. This method seems to be more of "add url with visit" unless it really does break if the url already exists?

Reply:
No breakage just adds another visit.